DOCSP-28126 OIDC mongosh options and connection (#267)
mongodb/docs-mongodb-shell: source/reference/options.txt

   - ``none``, the effect is the same as setting
     ``authMechanismProperties=CANONICALIZE_HOST_NAME:false``.

> .. option:: --oidcFlows

Reviewer: Let’s mention that this is a comma-separated list, i.e. you could have `--oidcFlows=auth-code,device-auth` (which I think we’d prefer over e.g. only `--oidcFlows=device-auth`, because we generally consider device-auth less secure).

Author: Done.

   Specifies OpenID Connect flows. The OpenID connect flows specify how the ``mongosh``
   interacts with identity provider for the authentication process. ``mongosh`` 
   supports the following OpenID Connect flows:

   .. list-table::
      :header-rows: 1
      :widths: 25 75

      * - OpenID Connect Flow
        - Description

      * - ``auth-code``
        - Default. ``mongosh`` redirects you to the identity provider log-in screen.

      * - ``device-auth``
        - ``mongosh`` provides you with a URL and code to finish authentication.    

.. option:: --oidcRedirectUri

   Specifies a URI where the identity provider redirects you after authentication.

>    The default is ``http://localhost:27097/redirect``.

Reviewer: The really important part here is that this needs to match the IDP configuration, so this should probably be mentioned in https://github.com/10gen/docs-mongodb-internal/pull/3335.

Author: Done.

.. option:: --oidcTrustedEndpoint

   Specifies a connection to a trusted endpoint that is not Atlas or localhost. 
   Only connect to hosts that you know are safe. 

.. option:: --browser

   Specifies the specific browser ``mongosh`` redirects you to when ``MONGODB-OIDC`` 

>    is enabled. 

Reviewer: We should mention that this is command will be run with the system shell (e.g. bash, zsh, etc.).

Author: Done.

   - ``none``, the effect is the same as setting
     ``authMechanismProperties=CANONICALIZE_HOST_NAME:false``.

.. option:: --oidcFlows

   Specifies OpenID Connect flows in a comma-separated list. 

>    The OpenID connect flows specify how ``mongosh``interacts with the identity 

Reviewer 2: ```suggestion
   The OpenID connect flows specify how ``mongosh`` interacts with the identity 
```

The space should fix the markup/rendering weirdness.

Author: Done.

>    The OpenID connect flows specify how ``mongosh``interacts with the identity 

Reviewer 2: (terminology) 
```suggestion
   The OpenID Connect flows specify how ``mongosh``interacts with the identity 
```

Author: Done.

   provider for the authentication process. ``mongosh`` supports the following 
   OpenID Connect flows:

   .. list-table::
      :header-rows: 1
      :widths: 25 75

      * - OpenID Connect Flow
        - Description

      * - ``auth-code``

>         - Default. ``mongosh`` redirects you to the identity provider log-in screen.

Reviewer: ```suggestion
        - Default. ``mongosh`` opens a browser and redirects you to the identity provider log-in screen.
```

Author: Done.

      * - ``device-auth``
        - ``mongosh`` provides you with a URL and code to finish authentication.

>           This is considered a less secure OpenID Connect flow.

Author: Done.

   To set ``device-auth`` as a fallback option to ``auth-code``, see the following 
   example:

   .. code-block:: bash 

      mongosh 'mongodb://localhost/' --authenticationMechanism MONGODB-OIDC --oidcFlows=auth-code,device-auth

.. option:: --oidcRedirectUri

   Specifies a URI where the identity provider redirects you after authentication.

>    The URI must match the configuration of the identity provider used by the server.

Reviewer: ```suggestion
   The URI must match the configuration of the identity provider.
```

(Having deja vu here, I might have already commented this somewhere else, this is not related to the configuration of the MongoDB server)

Reviewer: Ah, that was https://github.com/mongodb/docs-compass/pull/508/files#diff-e1fa5430d5ec6ced67a5a7b640244598b3cca80a7c07c48ac1222b4c0264a81eR109 – that comment is marked as resolved without changes but I still think it makes sense to update it both here and in the Compass docs

Author: Done.

   The default is ``http://localhost:27097/redirect``.

.. option:: --oidcTrustedEndpoint

   Specifies a connection to a trusted endpoint that is not Atlas or localhost. 

>    Only connect to hosts that you know are safe. 

Reviewer: ```suggestion
   Only use this option when connecting to servers that you trust.
```

(This is the wording from the Compass docs, I think it also makes sense to align that here)

Author: Done.

.. option:: --browser

>    Specifies the specific browser ``mongosh`` redirects you to when ``MONGODB-OIDC`` 

Reviewer 2: (wording and repitition)
```suggestion
   Specifies the browser ``mongosh`` redirects you to when ``MONGODB-OIDC`` 
```

Author: Done.
